templates/pwm.asg.element.json

```diff
@@ -0,0 +1,881 @@
+                                            "    - extras"
+                                        ]
+                                    ]
+                                },
+                                "group": "root",
+                                "mode": "000700",
+                                "owner": "root"
+                            },
+                            "/srv/salt/pillar/extras.sls": {
+                                "content": {
+                                    "Fn::Join": [
+                                        "",
+                                        [
+                                            "pwm:\n",
+                                            "  lookup:\n",
+                                            "    config_bucket: ",
+                                            {
+                                                "Ref": "ConfigBucketName"
+                                            }
+                                        ]
+                                    ]
+                                },
+                                "group": "root",
+                                "mode": "000700",
+                                "owner": "root"
+                            },
+                            "/etc/salt/minion.d/pwm.conf": {
+                                "content": {
+                                    "Fn::Join": [
+                                        "",
+                                        [
+                                            "file_client: local\n",
+                                            "file_roots: \n",
+                                            "  base:\n",
+                                            "  - /srv/salt/\n",
+                                            "pillar_roots: \n",
+                                            "  base:\n",
+                                            "  - /srv/salt/pillar/\n",
+                                            "hash_type: sha512\n",
+                                            "pillar_merge_lists: true\n",
+                                            "log_file: /var/log/salt/minion\n"
+                                        ]
+                                    ]
+                                },
+                                "group": "root",
+                                "mode": "000700",
+                                "owner": "root"
+                            }
+                        },
+                        "packages": {
+                            "yum": {
+                                "git": []
+                            }
+                        }
+                    },
+                    "d-adminusersetup": {
+                        "commands": {
+                            "10-cp_adminsshusermgmt": {
+                                "command": "cp /srv/salt/pwm/adminusermgmt.sh /usr/local/bin/adminusermgmt.sh"
```

Comment:
it's odd for this script to be distributed by the pwm formula, but managed by the cfn template instead of salt

Reply:
What's your recommendation for this? This isn't really a PWM configuration but at the same time it seems like we are avoiding putting configurations into the CloudFormation bucket as there is no way that the user would know they have to do this.

I would prefer this script exist in the CloudFormation portion as this is explicitly in the CFN side.

Reply:
If we move the file into this cfn-pwm repo, can we reference it with a local path? and do `aws cloudformation package` and `aws cloudformation deploy` then do the right thing, publishing the file to an s3 bucket, and converting the local path to the s3 reference?

If not, I'd say just leave it in the pwm-formula for now, and manage the location with salt rather than this cfn block...

Reply:
Looking at the documentation - https://docs.aws.amazon.com/cli/latest/reference/cloudformation/package.html - looks like I can not use `aws cloudformation package`. Moved the path and user/group blocks to be in the salt formula.